site/en/docs/extensions/mv3/manifest/content_scripts/index.md

```diff
@@ -0,0 +1,262 @@
+
+{% Label %}manifest.json{% endLabel %}
+
+```json/5
+{
+  ...
+  "content_scripts": [
+    {
+      "matches": ["https://*.nytimes.com/*"],
+      "exclude_globs": ["*science*"],
+      "js": ["content-script.js"]
+    }
+  ],
+  ...
+}
+```
+
+<div class="switcher">
+{% Compare 'better', 'Matches' %}
+```text
+https://history.nytimes.com
+https://.nytimes.com/music
+```
+{% endCompare %}
+
+{% Compare "worse", "Does not match" %}
+```text
+https://science.nytimes.com
+https://www.nytimes.com/science
+```
+{% endCompare %}
+</div>
+
+#### Advanced customization example {: #all-custom }
+
+{% Label %}manifest.json{% endLabel %}
+
+```json/5-7
+{
+  ...
+  "content_scripts": [
+    {
+      "matches": ["https://*.nytimes.com/*"],
+      "exclude_matches": ["*://*/*business*"],
+      "include_globs": ["*nytimes.com/???s/*"],
+      "exclude_globs": ["*science*"],
+      "js": ["content-script.js"]
+    }
+  ],
+  ...
+}
+```
+
+<div class="switcher">
+{% Compare 'better', 'Matches' %}
+```text
+https://www.nytimes.com/arts/index.html
+https://.nytimes.com/jobs/index.html
+```
+{% endCompare %}
+
+{% Compare "worse", "Does not match" %}
+```text
+https://science.nytimes.com
+https://www.nytimes.com/jobs/business
+https://www.nytimes.com/science
+```
+{% endCompare %}
+</div>
+
+{% endDetails %}
+
+## Frames {: #frames }
+
+The `"all_frames"` field specifies if the content script should be injected into all frames matching the specified URL requirements. If set to `false` it will only inject into the topmost frame. It can be used along with `"match_about_blank"` to inject into an `about:blank` frame. 
+
+To inject into other frames like `data:`, `blob:`, and `filesystem:`, set the `"match_origin_as_fallback"` to `true`. For details, see [Injecting in related frames][cs-inject-frames]
+
+`"all_frames"` Boolean
+: _Optional_. Defaults to `false`, meaning that only the top frame is matched.If specified true, it will inject into all frames, even if the frame is not the topmost frame in the tab. Each frame is checked independently for URL requirements, it won't inject into child frames if the URL requirements are not met.
+
+`"match_about_blank"`- Boolean
+: _Optional_. Defaults to `false`. Whether the script should inject into an `about:blank` frame where the parent URL matches one of the patterns declared in `"matches"`.
+
+`"match_origin_as_fallback"` - Boolean
+: _Optional_. Defaults to `false`. Whether the script should inject in frames that were created by a matching origin, but whose URL or origin may not directly match the pattern. These include frames with different schemes, such as `about:`, `data:`, `blob:`, and `filesystem:`.
+
+## Run time and execution environment {: #world-timings }
+
+By default, content scripts are injected when the DOM is complete (`"document_idle"`), meaning all its resources have finished loading, and live in a private isolated execution environment that isn't accessible to the page or other extensions.
```

Comment:
I could be wrong but (despite it being in the Chromium source) I'm not sure if "DOM is complete" is a good way of describing this accurately. It makes me think of [DOMContentLoaded](https://developer.mozilla.org/en-US/docs/Web/API/Document/DOMContentLoaded_event) which explicitly doesn't wait for images to load.